QueueInteropTransport.php

```diff
@@ -77,13 +78,18 @@ public function get(): iterable
             throw $e;
         }
 
-        return array(
-            $this->serializer->decode(array(
-                'body' => $interopMessage->getBody(),
-                'headers' => $interopMessage->getHeaders(),
-                'properties' => $interopMessage->getProperties(),
-            )),
-        );
+        /** @var Envelope $envelope */
+        $envelope = $this->serializer->decode(array(
+            'body' => $interopMessage->getBody(),
+            'headers' => $interopMessage->getHeaders(),
+            'properties' => $interopMessage->getProperties(),
+        ));
+
+        if ($envelope) {
+            $envelope = $envelope->with(new InteropMessageStamp($interopMessage));
+        }
+
+        return array($envelope);
     }
 
     /**
```

Comment:
It's unrelated, but `ack()`, `reject()` and `get()` should all throw a `TransportException` if something went wrong - e.g. https://github.com/symfony/symfony/blob/b92e4ed9d4e480ecfb5dfc0f1dd588b283c7e0a6/src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceiver.php#L63

Reply:
Also unrelated... `SendingMessageFailedException` should extend `TransportException`... as `send()` should also throw an instance of `TransportException` if there is a problem sending.

Reply:
That's a good point as well. Let me make a different PR for that though, to keep this one simple. Thank you for the review!